src/Chainweb/BlockHeaderDB/RestAPI/Client.hs

```diff
@@ -14,38 +23,106 @@
 module Chainweb.BlockHeaderDB.RestAPI.Client
 ( headerClient_
 , headerClient
+, headerClientContentType_
+, headerClientJson
+, headerClientJsonPretty
+, headerClientJsonBinary
+
 , headerPutClient_
 , headerPutClient
+, headerPutClientContentType_
+, headerPutClientJson
+, headerPutClientJsonPretty
+, headerPutClientBinary
+
 , hashesClient_
 , hashesClient
 , headersClient_
 , headersClient
+, headersClientContentType_
+, headersClientJson
+, headersClientJsonPretty
+
 , branchHashesClient_
 , branchHashesClient
+
 , branchHeadersClient_
 , branchHeadersClient
-, childHashesClient_
-, childHashesClient
-, childHeadersClient_
-, childHeadersClient
+, branchHeadersClientContentType_
+, branchHeadersClientJson
+, branchHeadersClientJsonPretty
 ) where
 
 import Control.Monad.Identity
 
+import Data.Kind
 import Data.Proxy
+import Data.Singletons
 
+import GHC.TypeLits
+
+import Servant.API
 import Servant.API (NoContent(..))
 import Servant.Client
 
 -- internal modules
+
+import Chainweb.BlockHash
+import Chainweb.BlockHeader
 import Chainweb.BlockHeaderDB (BlockHeaderDb)
 import Chainweb.BlockHeaderDB.RestAPI
-import Chainweb.ChainId
 import Chainweb.RestAPI.Orphans ()
 import Chainweb.TreeDB
 import Chainweb.Utils.Paging
 import Chainweb.Version
 
+-- -------------------------------------------------------------------------- --
+-- Utils
+
+-- Response Body Content Type
```

Comment:
Add comment that motivates the definitions in this section. Document the individual functions.